VideoLocker/src/main/java/org/edx/mobile/view/CourseCombinedInfoFragment.java

```diff
@@ -345,23 +340,71 @@ private void populateFacePile(){
 
     }
 
-    private View generateAnnouncementView(AnnouncementsModel model){
+    private void populateAnnouncements(List<AnnouncementsModel> announcementsList) {
+        if(announcementsList !=null && announcementsList.size()>0) {
+            hideEmptyAnnouncementMessage();
 
-        View convertView = inflater.inflate(R.layout.row_announcement_list, null);
+            // stop existing views before we remove them from the layout
+            for (AnnouncementView view : announcementViewList) {
+                view.stop();
+            }
+            announcementViewList.clear();
+            announcementContainer.removeAllViews();
+
+            for (AnnouncementsModel m : announcementsList) {
+                AnnouncementView announcementView = generateAnnouncementView(m);
+                announcementViewList.add(announcementView);
+
+                announcementContainer.addView(announcementView.getView());
```

Comment:
Merging into one webview maybe the right solution. For example, if
javascript opens a div-based dialog, multiple-webview approach may cause
the issue.  [partial showing of dialog...]

On Wed, Mar 11, 2015 at 12:05 PM, Akiva Leffert <notifications@github.com>
wrote:

> In
> VideoLocker/src/main/java/org/edx/mobile/view/CourseCombinedInfoFragment.java
> <https://github.com/edx/edx-app-android/pull/164#discussion_r26226193>:
>
> >
> > -        View convertView = inflater.inflate(R.layout.row_announcement_list, null);
> > +            // stop existing views before we remove them from the layout
> > +            for (AnnouncementView view : announcementViewList) {
> > +                view.stop();
> > +            }
> > +            announcementViewList.clear();
> > +            announcementContainer.removeAllViews();
> > +
> > +            for (AnnouncementsModel m : announcementsList) {
> > +                AnnouncementView announcementView = generateAnnouncementView(m);
> > +                announcementViewList.add(announcementView);
> > +
> > +                announcementContainer.addView(announcementView.getView());
>
> This might be complicated to make sure we keep your position after
> rotating from portrait to landscape once we do tablet support, but if you
> can make it work, that makes sense.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/164/files#r26226193>.
>